common/transport/amqp/test/_amqp_test.js

```diff
@@ -610,6 +610,59 @@ describe('Amqp', function () {
       });
     });
 
+    /*Tests_SRS_NODE_COMMON_AMQP_06_007: [While disconnecting, if the run down does not complete within 45 seconds, the code will be re-run with `forceDetach`es.]*/
+    it('Re-invokes the rundown if no responses on network within timeout', function(testCallback) {
+      this.clock = sinon.useFakeTimers();
+      var amqp = new Amqp();
+      var fakeConnection = new EventEmitter();
+      fakeConnection.name = 'connection';
+      var fakeConnectionContext = {connection: fakeConnection};
+      sinon.stub(amqp._rheaContainer, 'connect').callsFake(() => {process.nextTick(() => {amqp._rheaConnection.emit('connection_open', fakeConnectionContext)}); return fakeConnection});
+      var fakeSession = new EventEmitter();
+      fakeSession.name = 'session';
+      var fakeSessionContext = {connection: fakeConnection, session: fakeSession};
+      fakeConnection.create_session = sinon.stub().returns(fakeSession);
+      fakeConnection.close = () => {};
+      sinon.stub(fakeConnection, 'close').callsFake(() => {process.nextTick(() => {fakeConnection.emit('connection_close', fakeConnectionContext)})});
+      fakeSession.close = () => {};
+      sinon.stub(fakeSession, 'close').callsFake(() => {process.nextTick(() => {fakeSession.emit('session_close', fakeSessionContext)})});
+      fakeSession.open = () => {};
+      sinon.stub(fakeSession, 'open').callsFake(() => {process.nextTick(() => {fakeSession.emit('session_open', fakeSessionContext)})});
+
+      var fakeSender = {
+        detach: sinon.stub(),
+        forceDetach: sinon.stub()
+      };
+      var fakeReceiver = {
+        detach: sinon.stub(),
+        forceDetach: sinon.stub()
+      };
+
+      amqp.connect({uri: 'uri'}, function() {
+        amqp._senders.fake_sender_endpoint = fakeSender;
+        amqp._receivers.fake_receiver_endpoint = fakeReceiver;
+        amqp.disconnect(function(err) {
+          assert(fakeSender.detach.calledOnce);
+          assert(fakeSender.forceDetach.calledOnce);
+          assert(fakeReceiver.detach.calledOnce);
+          assert(fakeReceiver.forceDetach.calledOnce);
+          if (err) testCallback(err);
+          else {
+            testCallback();
+          }
```

Comment:
nit: linter would insist on having curly braces everywhere

Reply:
I love git blame.

Will fix.

---
In reply to: [261444583](https://github.com/Azure/azure-iot-sdk-node/pull/499#discussion_r261444583) [](ancestors = 261444583)